### PR TITLE
Define template unit for per-repo indexers (1.3.2)

### PR DESCRIPTION
**Pull request by leynos**
## Summary

This branch implements roadmap task `(1.3.2)` by adding the checked-in
`repovec-grepai@.service` template unit for per-repository grepai indexers and
extending the static systemd unit validator to cover the new template.

Roadmap task: `(1.3.2)`
Execplan: [docs/execplans/1-3-2-template-unit-for-per-repo-indexers.md](https://github.com/leynos/repovec-appliance/blob/b1700cf/docs/execplans/1-3-2-template-unit-for-per-repo-indexers.md)

The template runs as the `repovec` user, sets `HOME=/var/lib/repovec`, depends
on Qdrant and `repovecd`, keeps output in journald, and includes conservative
systemd sandboxing directives. Concrete indexer instance reconciliation remains
in roadmap item `3.2.1`.

## Review walkthrough

- Start with [packaging/systemd/repovec-grepai@.service](https://github.com/leynos/repovec-appliance/blob/b1700cf/packaging/systemd/repovec-grepai%40.service) for the shipped template contract.
- Review [crates/repovec-core/src/appliance/systemd_units/mod.rs](https://github.com/leynos/repovec-appliance/blob/b1700cf/crates/repovec-core/src/appliance/systemd_units/mod.rs) and [crates/repovec-core/src/appliance/systemd_units/parsed.rs](https://github.com/leynos/repovec-appliance/blob/b1700cf/crates/repovec-core/src/appliance/systemd_units/parsed.rs) for the validator and parser split.
- Check [crates/repovec-core/src/appliance/systemd_units/tests.rs](https://github.com/leynos/repovec-appliance/blob/b1700cf/crates/repovec-core/src/appliance/systemd_units/tests.rs) and [crates/repovec-core/tests/systemd_units_bdd.rs](https://github.com/leynos/repovec-appliance/blob/b1700cf/crates/repovec-core/tests/systemd_units_bdd.rs) for unit and behavioural coverage.
- Finish with [docs/repovec-appliance-technical-design.md](https://github.com/leynos/repovec-appliance/blob/b1700cf/docs/repovec-appliance-technical-design.md), [docs/users-guide.md](https://github.com/leynos/repovec-appliance/blob/b1700cf/docs/users-guide.md), [docs/developers-guide.md](https://github.com/leynos/repovec-appliance/blob/b1700cf/docs/developers-guide.md), and [docs/roadmap.md](https://github.com/leynos/repovec-appliance/blob/b1700cf/docs/roadmap.md) for the documented boundary and roadmap completion note.

## Validation

- `make build`: passed.
- `make check-fmt`: passed.
- `make typecheck`: passed.
- `make lint`: passed.
- `make test`: passed; 240 nextest tests and 33 doctests passed.
- `make markdownlint`: passed.
- `make nixie`: passed.
- `coderabbit review --agent`: passed with zero findings after implementation and documentation milestones before this review-feedback pass.

## Notes

`make fmt` remains blocked by `mdformat-all` invoking a stricter `markdownlint`
command that reports unrelated repository-wide line-length and reference issues
in existing documentation. The changed docs pass the repository's explicit
`make markdownlint` target, and incidental formatter churn was not retained.

## References

- Lody session: https://lody.ai/leynos/sessions/7d87a312-c490-4807-a011-ae802ec107e9

### CODE REVIEW

**Comment by pandalump** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai resume

File: crates/repovec-core/src/appliance/systemd_units/tests.rs [hunk 1]
```diff
@@ -52,6 +52,24 @@ pub(super) enum ValidationScenario {
...
+    WrongGrepaiTemplateType,
+    WrongGrepaiTemplateExecStart,
+    GrepaiTemplateWrongUser,
+    GrepaiTemplateMissingGroup,
+    GrepaiTemplateWrongWorkingDirectory,
+    GrepaiTemplateMissingEnvironment,
+    GrepaiTemplateWrongRestartPolicy,
+    GrepaiTemplateWrongRestartDelay,
+    GrepaiTemplateLogsStdoutToFile,
+    GrepaiTemplateLogsStderrToFile,
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/repovec-core/src/appliance/systemd_units/tests.rs:73]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/execplans/1-3-2-template-unit-for-per-repo-indexers.md [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+  where applicable.
+- Property tests, Kani, or Verus are not required for this finite static
+  packaging contract. Add them only if implementation introduces a real
+  invariant over an open input space.
+- Documentation must use en-GB Oxford spelling, wrap Markdown paragraphs at
+  80 columns, and follow `docs/documentation-style-guide.md`.
+- Before each implementation commit and before any CodeRabbit review, run the
+  applicable gates sequentially with `tee` logs under `/tmp`: at minimum
+  `make check-fmt`, `make typecheck`, `make lint`, and `make test`. Run
+  `make fmt`, `make markdownlint`, and `make nixie` after documentation changes.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/execplans/1-3-2-template-unit-for-per-repo-indexers.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/execplans/1-3-2-template-unit-for-per-repo-indexers.md [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+  Date/Author: 2026-06-02T04:58:12+02:00 / Codex.
+
+## Outcomes & Retrospective
+
+No implementation has happened yet. This draft identifies the work, validation
+strategy, risks, and approval gate for roadmap item `1.3.2`.
+
+After implementation, update this section with the committed unit template,
+validator changes, test results, CodeRabbit outcome, documentation updates, and
+any deviations from the plan.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/execplans/1-3-2-template-unit-for-per-repo-indexers.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/execplans/1-3-2-template-unit-for-per-repo-indexers.md [hunk 4]
```diff
@@ -0,0 +1,632 @@
...
+- `packaging/systemd/repovec-grepai@.service` exists and contains no bespoke
+  log-file routing.
+- The checked-in template runs as `repovec:repovec` with
+  `HOME=/var/lib/repovec`.
+- The checked-in template starts `grepai watch` without shell wrappers,
+  pipelines, or redirection.
+- The checked-in template is ordered after Qdrant and tied to
+  `repovec.target`.
+- The static validator rejects missing or incorrect template identity,
+  dependency, command, environment, journald, and install directives.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/execplans/1-3-2-template-unit-for-per-repo-indexers.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/repovec-appliance-technical-design.md [hunk 5]
```diff
@@ -342,6 +344,28 @@ uses. `repovecd.service` declares both `Requires=qdrant.service` and
...
+`repovec-grepai@.service` is a systemd template for future concrete indexer
+instances. Each instance runs `/usr/bin/grepai watch` as the `repovec` user and
+group, sets `HOME=/var/lib/repovec`, and uses
+`WorkingDirectory=/var/lib/repovec/worktrees/%I`. The `%I` value is the systemd
+instance identifier; roadmap item `3.2.1` owns the final mapping from
+repository and branch identity to a concrete, systemd-safe instance name. The
+template declares `Requires=qdrant.service repovecd.service`,
+`After=qdrant.service repovecd.service`, `PartOf=repovec.target`, and
+`WantedBy=repovec.target`, so manually enabled instances are bound to the
+appliance target and stop with it. The template keeps stdout and stderr in
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/repovec-appliance-technical-design.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/users-guide.md [hunk 6]
```diff
@@ -218,6 +216,13 @@ and `cloudflared.service`. The Qdrant service name is the generated systemd
 unit from the installed `/etc/containers/systemd/qdrant.container` Quadlet;
 dependent services must use `qdrant.service`.
 
+`repovec-grepai@.service` is the template used for future per-repository
+indexer instances. It runs `grepai watch` as the `repovec` user, sets
+`HOME=/var/lib/repovec`, works in `/var/lib/repovec/worktrees/<instance>`, and
+writes stdout and stderr to journald. Later reconciliation work creates and
+manages concrete instances; operators should not expect installing the template
+alone to start indexers.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/users-guide.md:224]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### crates/repovec-core/src/appliance/systemd_units/mod.rs (1)

`8-16`: _⚠️ Potential issue_ | _🟡 Minor_ | _⚡ Quick win_
**Synchronize the module Rustdoc with the grepai template contract.**
Update the entry-point and contract-scope lists: `validate_checked_in_systemd_units`

now validates four embedded assets, and the public

`validate_systemd_units_with_grepai_template` entry point plus grepai

`ExecStart`/template directives are missing from the module overview.

##### Suggested Rustdoc update

```diff
-//! - [`validate_checked_in_systemd_units`] validates the three embedded unit
-//!   assets shipped in the repository.
+//! - [`validate_checked_in_systemd_units`] validates the embedded target,
+//!   daemon service, and grepai indexer template assets shipped in the
+//!   repository.
 //! - [`validate_and_trace_checked_in_units`] validates the checked-in unit set
 //!   and emits the daemon startup success trace.
 //! - [`validate_systemd_units`] validates caller-supplied unit text. Use it in
 //!   tests or tooling that needs to analyse unit contents sourced outside the
 //!   checked-in files.
+//! - [`validate_systemd_units_with_grepai_template`] validates caller-supplied
+//!   unit text plus the grepai indexer template contract.
@@
-//! - `ExecStart=` executable paths for `repovecd` and `repovec-mcpd`.
+//! - `ExecStart=` executable paths for `repovecd`, `repovec-mcpd`, and
+//!   `grepai watch`.
 //! - `[Service]` identity and runtime-directory directives: `User=`, `Group=`,
 //!   `WorkingDirectory=`, and `Environment=HOME=`.
```

As per coding guidelines, “Document public APIs in Rust using Rustdoc comments” and “Ensure that any API or behavioural changes are reflected in the documentation in `docs/`.”
Also applies to: 28-39

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@crates/repovec-core/src/appliance/systemd_units/mod.rs` around lines 8 - 16,
The module-level Rustdoc comments in the systemd_units module are out of sync
with the actual implementation. Update the documentation comment starting at
line 8 to reflect that validate_checked_in_systemd_units now validates four
embedded unit assets instead of three, and add documentation for the missing
public entry point validate_systemd_units_with_grepai_template along with its
purpose. Additionally, include information about the grepai ExecStart and
template directives in the module overview. Also update the related
documentation block mentioned at lines 28-39 to maintain consistency across all
API documentation in this module.

_Source: Coding guidelines_

**Comment by leynos** [comment]
@coderabbitai resume

**Comment by leynos** [comment]
@coderabbitai lody is the agent hosting platform we use. Robots live in a house in Helsinki, powered by lody. With lots of boardgames, knitting and coffee. Happy robots write happy code!

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### crates/repovec-core/src/appliance/systemd_units/mod.rs (1)

`8-16`: _⚠️ Potential issue_ | _🟡 Minor_ | _⚡ Quick win_
**Synchronize the module Rustdoc with the grepai template contract.**
Update the entry-point and contract-scope lists: `validate_checked_in_systemd_units`

now validates four embedded assets, and the public

`validate_systemd_units_with_grepai_template` entry point plus grepai

`ExecStart`/template directives are missing from the module overview.

##### Suggested Rustdoc update

```diff
-//! - [`validate_checked_in_systemd_units`] validates the three embedded unit
-//!   assets shipped in the repository.
+//! - [`validate_checked_in_systemd_units`] validates the embedded target,
+//!   daemon service, and grepai indexer template assets shipped in the
+//!   repository.
 //! - [`validate_and_trace_checked_in_units`] validates the checked-in unit set
 //!   and emits the daemon startup success trace.
 //! - [`validate_systemd_units`] validates caller-supplied unit text. Use it in
 //!   tests or tooling that needs to analyse unit contents sourced outside the
 //!   checked-in files.
+//! - [`validate_systemd_units_with_grepai_template`] validates caller-supplied
+//!   unit text plus the grepai indexer template contract.
@@
-//! - `ExecStart=` executable paths for `repovecd` and `repovec-mcpd`.
+//! - `ExecStart=` executable paths for `repovecd`, `repovec-mcpd`, and
+//!   `grepai watch`.
 //! - `[Service]` identity and runtime-directory directives: `User=`, `Group=`,
 //!   `WorkingDirectory=`, and `Environment=HOME=`.
```

As per coding guidelines, “Document public APIs in Rust using Rustdoc comments” and “Ensure that any API or behavioural changes are reflected in the documentation in `docs/`.”
Also applies to: 28-39

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@crates/repovec-core/src/appliance/systemd_units/mod.rs` around lines 8 - 16,
The module-level Rustdoc comments in the systemd_units module are out of sync
with the actual implementation. Update the documentation comment starting at
line 8 to reflect that validate_checked_in_systemd_units now validates four
embedded unit assets instead of three, and add documentation for the missing
public entry point validate_systemd_units_with_grepai_template along with its
purpose. Additionally, include information about the grepai ExecStart and
template directives in the module overview. Also update the related
documentation block mentioned at lines 28-39 to maintain consistency across all
API documentation in this module.

_Source: Coding guidelines_

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already. (Treat documentation and validation coverage as in scope).

### ❌ Failed checks (1 warning)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Property / Proof) | ⚠️ Warning | The PR introduces a systemd unit parser (ParsedUnit) with invariants around whitespace normalisation, tokenisation of multi-value directives, and value matching, yet tests only two parser error cas... | Recommend proptest-based property tests covering: whitespace permutations in directives, multi-value directive tokenisation edge cases, and value matching invariants across section/key combinations. |

**Comment by pandalump** [comment]
@coderabbitai review
